src/utils/DateUtils.ts

```diff
@@ -29,138 +155,231 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
   predefinedFormat: string = undefined;
 }
 
+/**
+ * The `DateUtils` class exposes methods to convert strings, numbers and date objects to standard ISO 8601 Date objects,
+ * using the correct culture, language and format. It also offers methods to convert date objects to strings.
+ */
 export class DateUtils {
-  static convertFromJsonDateIfNeeded(date: string): Date;
-  static convertFromJsonDateIfNeeded(date: number): Date;
-  static convertFromJsonDateIfNeeded(date: Date): Date;
+  // This function is used to call convertToStandardDate for legacy reasons. convertFromJsonDateIfNeeded was refactored to
+  // convertToStandardDate, which would be a breaking change otherwise.
   static convertFromJsonDateIfNeeded(date: any): Date {
+    return DateUtils.convertToStandardDate(date);
+  }
+
```

Comment:
I'd write:

> Tries to parse an argument of any type to a standard Date object.
> @param date The value to parse. Can be of any type (string, number, Date, etc.).
> @returns {any} The parsed Date object, or `Invalid Date` if the `date` argument was not recognized as a valid date.

In general, try to avoid using contractions (e.g., "wasn't") in public documentation. It's okay in comments, though :)